src/ashuffle.c

```diff
@@ -161,7 +161,22 @@ int build_songs_mpd(struct mpd_connection * mpd,
  * songs to the queue */
 void queue_random_song(struct mpd_connection * mpd,
                        struct shuffle_chain * songs) {
-    if (mpd_run_add(mpd, shuffle_pick(songs)) != true) { mpd_perror(mpd); }
+    const char * item = shuffle_pick(songs);
+    const char * suffix = item;
+    /* find last dot */
+    for(int n = 0; item[n]; n++) {
+        if(item[n] == '.') {
+            suffix = &item[n+1];
+        }
+    }
+    bool is_playlist = ( \
```

Comment:
No need for the ternary operator, just use logical or between the `strcmp(...) == 0` checks.

Reply:
That would result in superfluous checks once we determine this is actually a playlist from the suffix.

Reply:
Logical OR (`||`) is short-cutting in C (as in most languages), so no superfluous checks will take place.